src/main/java/org/springframework/data/couchbase/core/convert/DateConverters.java

```diff
@@ -52,9 +56,18 @@ private DateConverters() {
   public static Collection<Converter<?, ?>> getConvertersToRegister() {
     List<Converter<?, ?>> converters = new ArrayList<Converter<?, ?>>();
 
-    converters.add(DateToLongConverter.INSTANCE);
+    boolean useISOStringConverterForDate = Boolean.parseBoolean(
+            System.getProperty("org.springframework.data.couchbase.useISOStringConverterForDate", "false"));
+
+    if (useISOStringConverterForDate) {
+      converters.add(DateToStringConverter.INSTANCE);
+      converters.add(StringToDateConverter.INSTANCE);
+    } else {
+      converters.add(DateToLongConverter.INSTANCE);
+      converters.add(NumberToDateConverter.INSTANCE);
```

Comment:
You bring up a very good point 👍 If the property is set, reading converter should indeed accept both long and string.